client/www/scripts/modules/manager/manager.controllers.js

```diff
@@ -609,6 +616,59 @@ Manager.controller('ManagerMainController', [
       }
     };
 
+    function updateHostEnv(host, env) {
+      var deferred = $q.defer();
+      var command =  {
+        cmd: 'env-set',
+        env: env
+      };
+
+      if (host) {
+        host.action(command, function(err, res) {
+          if (err) {
+            deferred.reject(err);
+          } else {
+            deferred.resolve(env);
+          }
+        });
+      }
```

Comment:
yeah, promise chaining only works with promisified functions, and returning instead of resolving only works inside $http or $q, from what I can tell.

You may have to do it the way you're doing it, i just noted that the line:

`if ( !host ) return` should actually call `def.reject(err);` or `def.resolve()` depending on what you want to do so it doesn't break the promise chain.